khal/aux.py

```diff
@@ -392,163 +444,85 @@ def generate_random_uid():
+                                 (datetimefstr, locale['longdateformat'])]:
+                try:
+                    until_date = fun(until.split(' '), dformat)
+                    break
+                except ValueError:
+                    pass
+            if until_date is None:
+                logger.fatal("Cannot parse until date: '{}'\nPlease have a look "
+                             "at the documentation.".format(until))
+                raise FatalError()
+            rrule_settings['until'] = until_date
+
+        return rrule_settings
     else:
-        if not dtime_list:
-            logger.fatal('No event summary provided, aborting.')
-            raise FatalError
-        try:
-            # next element is a valid Olson db timezone string
-            dtstart = pytz.timezone(dtime_list[0]).localize(dtstart)
-            dtend = pytz.timezone(dtime_list[0]).localize(dtend)
-            dtime_list.pop(0)
-        except (pytz.UnknownTimeZoneError, UnicodeDecodeError):
-            dtstart = locale['default_timezone'].localize(dtstart)
-            dtend = locale['default_timezone'].localize(dtend)
+        logger.fatal("Invalid value for the repeat option. \
+                Possible values are: daily, weekly, monthly or yearly")
+        raise FatalError()
 
-    event = icalendar.Event()
-    text = ' '.join(dtime_list)
-    if not description or not location:
-        summary = text.split(' :: ', 1)[0]
-        try:
-            description = text.split(' :: ', 1)[1]
-        except IndexError:
-            pass
-    else:
-        summary = text
 
-    if description:
-        event.add('description', description)
-    if location:
-        event.add('location', location)
-    if categories:
-        event.add('categories', categories)
-    if repeat and repeat != "none":
-        if repeat in ["daily", "weekly", "monthly", "yearly"]:
-            rrule_settings = {'freq': repeat}
-            if until:
-                until_date = None
-                for fun, dformat in [(datetimefstr, locale['datetimeformat']),
-                                     (datetimefstr, locale['longdatetimeformat']),
-                                     (timefstr, locale['timeformat']),
-                                     (datetimefstr, locale['dateformat']),
-                                     (datetimefstr, locale['longdateformat'])]:
-                    try:
-                        until_date = fun(until, dformat)
-                        break
-                    except ValueError:
-                        pass
-                if until_date is None:
-                    logger.fatal("Cannot parse until date: '{}'\nPlease have a look "
-                                 "at the documentation.".format(until))
-                    raise FatalError()
-                rrule_settings['until'] = until_date
-
-            event.add('rrule', rrule_settings)
+def eventinfofstr(info_string, locale, default_timedelta=None,
```

Comment:
please write a docstring that describes what those parameters do and what you expect `info_string` to look like